pkg/repository/udmrepo/kopialib/backend/file_system.go

```diff
@@ -62,6 +63,13 @@ func (c *FsBackend) Connect(ctx context.Context, isCreate bool, logger logrus.Fi
 	if !filepath.IsAbs(c.options.Path) {
 		return nil, errors.Errorf("filesystem repository path is not absolute, path: %s", c.options.Path)
 	}
+
+	if !isCreate {
+		if _, err := os.Stat(c.options.Path); err != nil {
+			return nil, ErrStoreNotExist
+		}
```

Comment:
```
  if !isCreate {
      if _, err := os.Stat(c.options.Path); err != nil {
          if os.IsNotExist(err) {
              return nil, ErrStoreNotExist
          }
          return nil, errors.Wrap(err, "failed to stat repository path")
      }
  }
```
Do we want ErrStoreNotExist err to only return for IsNotExist? or also for other errors such as permissions, I/O

Reply:
For notExist only.
Permission or IO errors should be regarded as a failure of repo access and handled by repo connection.